app/include/drivers/behavior.h

```diff
@@ -23,6 +23,51 @@
  * (Internal use only.)
  */
 
+#if IS_ENABLED(CONFIG_ZMK_BEHAVIOR_METADATA)
+
+enum behavior_parameter_standard_domain {
+    BEHAVIOR_PARAMETER_STANDARD_DOMAIN_NULL = 0,
+    BEHAVIOR_PARAMETER_STANDARD_DOMAIN_HID_USAGE = 1,
+    BEHAVIOR_PARAMETER_STANDARD_DOMAIN_LAYER_INDEX = 2,
+    // TODO: Implement along with underglow RGB refactor
+    // BEHAVIOR_PARAMETER_STANDARD_DOMAIN_RGB = 3,
+};
+
+struct behavior_parameter_value_metadata {
+    char *friendly_name;
+
+    union {
+        uint32_t value;
+        struct {
+            int32_t min;
+            int32_t max;
+        } range;
+
+        enum behavior_parameter_standard_domain standard;
+    };
+
+    enum {
+        BEHAVIOR_PARAMETER_VALUE_METADATA_TYPE_VALUE = 0,
```

Comment:
Do we need "metadata" here? This is the type of the parameter value, not the type of the metadata itself, so this naming could be a little confusing.

Reply:
Shortened.